pulsar-client-go/pulsar/producer.go

```diff
@@ -125,6 +125,12 @@ type ProducerOptions struct {
 	// release in order to be able to receive messages compressed with ZSTD.
 	CompressionType
 
+	// Declare the schema of the data that this consumer will be accepting.
+	//
+	// The schema will be checked against the schema of the topic, and the
+	// consumer creation will fail if it's not compatible.
+	SchemaInfo
```

Comment:
To keep consistency with consumer options, we should keep it as `Schema SchemaInfo`

Reply:
Actually, `SchemaInfo` it's just a part of whole equation. It represents the schema definition, but, in this case since it's struct it doesn't have any behavior associated.

When creating a producer/consumer we should pass a "Schema" interface, that will encapsulate both the serde and will be able to provide a schemainfo.